src/Symfony/Component/FeatureFlag/FeatureChecker.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\FeatureFlag;
+
+use Symfony\Component\FeatureFlag\Provider\ProviderInterface;
+
+/**
+ * @experimental
+ */
+final class FeatureChecker implements FeatureCheckerInterface
+{
+    private array $cache = [];
```

Comment:
Question regarding « Once the feature's value is resolved, it is stored to make it immutable » (PR desc)

Isnt necessary then to implement the reset interface of the kernel to purge this cache ?